detox/android/detox/src/full/java/com/wix/detox/espresso/matcher/ViewMatchers.kt

```diff
@@ -44,3 +51,30 @@ private class IsAssignableFromMatcher(private val clazz: Class<*>) : TypeSafeMat
         description.appendText("is assignable from class: $clazz")
     }
 }
+
+fun toHaveSliderPosition(expectedValue: Double, tolerance: Double): Matcher<View?> {
+    return object : BoundedMatcher<View?, ReactSlider>(ReactSlider::class.java) {
+        override fun describeTo(description: Description) {
+            description.appendText("expected: $expectedValue")
+        }
+
+        override fun matchesSafely(slider: ReactSlider?): Boolean {
+            val currentProgress = slider?.progress
+
+            if (currentProgress != null) {
+                val realProgress = slider.toRealProgress(currentProgress)
+                val currentPctFactor = slider.max/currentProgress.toDouble()
+                val realTotal = realProgress * currentPctFactor
+                val actualValue = realProgress/ realTotal
+
+                if (tolerance > 0) {
+                    return Math.abs(actualValue - expectedValue) < tolerance
+                }
+
+                return actualValue == expectedValue
```

Comment:
Could we set `tolerance` to 1.0 (i.e. max) and use the previous equation to match both cases?